plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3select/TrinoS3SelectClient.java

```diff
@@ -82,4 +103,118 @@ public boolean isRequestComplete()
     {
         return requestComplete;
     }
+
+    /**
+     * Below classes are required for compatibility between AWS Java SDK 1.x and 2.x
```

Comment:
What does compatibility mean here? Since we're not supporting the 1.x SDK anymore, could we simplify the code?

Reply:
The SDKv2 does not return an `InputStream` for the data like SDK v1. I think the Hadoop LineReader above this layer requires an InputStream. This glue code takes the event stream and converts it into an InputStream. Maybe in the new implementation, we can remove the need for an InputStream and directly use the event stream?

Reply:
This may be out of our hands, but it really feels like this is something the SDK should be able to provide for us. Getting object content back as an input stream seems like a common enough use case to have it be baked in. It's not like InputStreams are a Hadoop specific thing.